Source/WebGPU/WebGPU/Device.h

```diff
@@ -241,6 +242,17 @@ class Device : public WGPUDeviceImpl, public ThreadSafeRefCountedAndCanMakeThrea
 #if HAVE(COREVIDEO_METAL_SUPPORT)
     RetainPtr<CVMetalTextureCacheRef> m_coreVideoTextureCache;
 #endif
-};
+} SWIFT_SHARED_REFERENCE(retainDevice, releaseDevice);
```

Comment:
I guess I don't understand what is the difference to passing ` WTF::retainThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr, WTF::releaseThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr` to the macro instead

Reply:
then you don't need the custom inline functions in the header

Reply:
Interop doesn not support templated retain release functions. Hence the wrapper.